userland/src/parameters.cpp

```diff
@@ -0,0 +1,74 @@
+#include <algorithm>
+#include <chrono>
+#include <cmath>
+#include <iostream>
+#include <string>
+#include <thread>
+
+#include <rclcpp/rclcpp.hpp>
+
+template <typename ParameterTypeT>
+ParameterTypeT
+wait_for_future(rclcpp::node::Node::SharedPtr & node, std::shared_future<ParameterTypeT> & future)
+{
+  std::future_status status;
+  do {
+    rclcpp::spin_some(node);
+    status = future.wait_for(std::chrono::milliseconds(100));
```

Comment:
What's the rational for waiting here? Wouldn't it be better to wait for something to happen in the spin and then do a non-blocking check and then spin again until the future is ready? It seems to me that the future will never become ready except during a spin, so waiting outside of the spin doesn't do anything other than slow down the busy loop polling of `spin_some`.

Instead we should get a blocking version of `spin_some` which can block until something happens and the do a non-blocking check of the future's status.